### DISCUSSION TITLE
Cannot Get the BlackPIll to work with ST7735, Please Help.

### DISCUSSION THREAD

**Discussion by thuvasooriya** [Bodmer/TFT_eSPI#3017] [Compatible displays and setup files]
Output for running read setup information

```
TFT_eSPI ver = 2.5.34
Processor    = STM32
Transactions = Yes
Interface    = SPI
Display driver = 7735
Display width  = 128
Display height = 160

R0 x offset = 2
R0 y offset = 1

>>>>> Note: STM32 pin references above D15 may not reflect board markings <<<<<
TFT_CS   = D-64
TFT_DC   = D-63
TFT_RST  = D-62

TFT_BL           = D1
TFT_BACKLIGHT_ON = HIGH

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 27.00
```

I've tried so many variations. Still I might have missed something.
Please help me on how to set this up. and what am I doing wrong.
I'm using platformIO with stm32duino and this is the display I bought. https://github.com/WeActStudio/WeAct-ST7735/tree/master
When I leave Backlight wire not connected, I can see that the display is doing stuff that I programmed it to do without a blacklight. If I use pwm for backlight it just results in white screen.

* The display works with Adafruit library on Arduino.

**Comment by Bodmer** [comment]
Try with the BL signal set high (or low) so LEDs are on. Does the display work the? If so then it sounds like a power supply problem. Are you running the display from 5V?

**Reply by thuvasooriya** [reply]
Awesome That was a great advice... Thankyou very much. I've seen so many guides but I haven't seen anywhere (as far as I remember) that mentions to connect the BL to low (ground to be exact). Just connecting the BL to Ground works like a charm in my black pill. Thank you very much.

**Comment by thuvasooriya** [comment]
Even though that solved that issue I'm hoping you'll consider to put some advice regarding another issue since it's a discussion. I'm trying the display on another board STM32F405 from WeAct (Similar to Adafruit feather), and the same configurations(configurations that worked for the blackpill) are not working for that board. the screen is just white. cannot observe anything. Any help regarding this would be awesome. @Bodmer  Thanks in advance.

**Reply by Bodmer** [reply]
Assuming you are using SPI port 1 pins (PA5/6/7) for the SPI pins then I would expect it to work. Caveat is that I do not have that processor variant and have not tested it, however it should be compatible with other STM32F4xx processors.

**Reply by thuvasooriya** [reply]
Yup, after spending some more time, I tried the 2nd SPI and it works I don't know why it doesn't work with the 1st SPI interface. I only needed to change the SDL, SCK pins and it works. This is really strange because I haven't occupied the IO with anything else and only test code is running. I thought the speed of the hardware SPI might do something to TFT, but we are defining it. So I couldn't figure it out. If you happen to know any reason for the displays to only work in one interface, please do tell. Thankyou for taking the time to reply. 🥂

**Reply by Bodmer** [reply]
It might be the SPI speed as you suggest. I think port 2 limits the speed to 24MHz, whereas port 1 is 48MHz maximum. Typically ST7735 displays work up to about 27MHz. Did you try port 1 with a slower speed?

**Reply by thuvasooriya** [reply]
Thanks for the reply, Everything works fine !, it seems like a mistake on my part. I only defined the SPI port 1 on black pill and didn't define the TFT_MOSI values yet it still worked. I'm not sure it's supposed to do that. because when I did the same thing in F405 it didn't work but it worked when I defined the pins. So yes defining TFT_MOSI and the required 3 pins work on any SPI. not sure about how black pill auto detects that stuff. Thanks anyway. ✌🏽
